src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -909,12 +909,25 @@ private NuGetProject CreateNuGetProject(Project envDTEProject, INuGetProjectCont
 
             // read MSBuild property RestoreProjectStyle which can be set to any NuGet project sytle
             // and pass it on to NugetFactory which can pass it to each NuGet project provider to consume.
-            var restoreProjectStyle = VsHierarchyUtility.GetMSBuildProperty(VsHierarchyUtility.ToVsHierarchy(envDTEProject), "RestoreProjectStyle");
+            var restoreProjectStyle = VsHierarchyUtility.GetMSBuildProperty(VsHierarchyUtility.ToVsHierarchy(envDTEProject), 
+                ProjectSystemProviderContext.RESTORE_PROJECT_STYLE);
+
+            var targetFramework = VsHierarchyUtility.GetMSBuildProperty(VsHierarchyUtility.ToVsHierarchy(envDTEProject), 
+                ProjectSystemProviderContext.TARGET_FRAMEWORK);
+
+            var targetFrameworks = VsHierarchyUtility.GetMSBuildProperty(VsHierarchyUtility.ToVsHierarchy(envDTEProject), 
+                ProjectSystemProviderContext.TARGET_FRAMEWORKS);
+
+            var msBuildProperties = new Dictionary<string, string> {
```

Comment:
We do NullOrEmptyCheck when we consume them. We anyway need to do a Null check when we TryGetValue. Checking them here will result in 2 Null checks.